crates/router/src/bin/router.rs

```diff
@@ -8,18 +8,22 @@ use router::{
 async fn main() -> ApplicationResult<()> {
     // get commandline config before initializing config
     let cmd_line = <CmdLineConf as clap::Parser>::parse();
-    if let Some(Subcommand::GenerateOpenapiSpec) = cmd_line.subcommand {
-        let file_path = "openapi/generated.json";
-        #[allow(clippy::expect_used)]
-        std::fs::write(
-            file_path,
-            <router::openapi::ApiDoc as utoipa::OpenApi>::openapi()
-                .to_pretty_json()
-                .expect("Failed to generate serialize OpenAPI specification as JSON"),
-        )
-        .expect("Failed to write OpenAPI specification to file");
-        println!("Successfully saved OpenAPI specification file at '{file_path}'");
-        return Ok(());
+    #[cfg(feature = "openapi")]
```

Comment:
Could you please add this feature gate on the `CmdLineConf.subcommand` field as well? It's so that when that feature is disabled, the subcommand does not show up on the help message.

Reply:
Done. Have added in the latest commit